internal/tenant/index.go

```diff
@@ -0,0 +1,79 @@
+package tenant
+
+import (
+	"context"
+	"iter"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	proto "github.com/sourcegraph/zoekt/cmd/zoekt-sourcegraph-indexserver/protos/sourcegraph/zoekt/configuration/v1"
+	"github.com/sourcegraph/zoekt/internal/tenant/internal/tenanttype"
+)
+
+// ContextIndexDir returns a context and index dir for the given tenant ID.
+func ContextIndexDir(tenantID int, repoDir string) (context.Context, string) {
+	if !EnforceTenant() {
+		// Default to tenant 1 if enforcement is disabled.
+		return tenanttype.WithTenant(context.Background(), 1), repoDir
+	}
+	return tenanttype.WithTenant(context.Background(), tenantID), filepath.Join(repoDir, TenantsDir, strconv.Itoa(tenantID))
+}
+
+// HttpExtraHeader returns header we send to gitserver given a tenant context.
+func HttpExtraHeader(ctx context.Context) string {
+	key := headerKeyTenantID + ": "
+	if !EnforceTenant() {
+		return key + "1"
+	}
+	tnt, err := tenanttype.FromContext(ctx)
```

Comment:
in sg/sg, we pprof log the callers of FromContext as missing_tenant to generate profiles where we can spot missing tenant errors quickly. Would that be worth adding here as well?